examples/all-clusters-app/all-clusters-common/gen/CHIPClustersObjc.mm

```diff
@@ -3805,8 +3806,8 @@ - (void)commissioningComplete:(ResponseHandler)completionHandler
         completionHandler([CHIPError errorForCHIPErrorCode:err], nil);
     }
 }
-- (void)setFabric:(char *)fabricId
-         fabricSecret:(char *)fabricSecret
+- (void)setFabric:(chip::ByteSpan)fabricId
```

Comment:
This does not match the declaration in the header, so this part does not make sense to me.  @sagar-apple what's the sanest thing to do here?

Reply:
Depends on the usage... What do we expect the caller to know about fabrics? i.e imagine someone needing to call this API, do they need to provide a `NSString *` like the header says or do we need a `ByteSpan`? What information/utilities do they have access to generate a valid input here for FabricId?

If we stick to Objc Types like NSString or NSData for raw bytes, this method must perform conversion to the types that the underlying cpp wants.

Reply:
Oh, this is calling _into_ C++, not being called _from_ C++?  I should have read more of the context!

In that case, things are pretty straightforward, I think.  The arg type should be `NSData*` in both the header and the .mm.  The impl code in the .mm should then do something like this when doing the c++ call:
```
  chip::ByteSpan((const uin8_t*)fabricId.bytes, fabricId.length)
```
and similar for the other cases like this.

Reply:
Yes, NSData should be used here, thanks.